src/Microsoft.Health.TaskManagement/JobHosting.cs

```diff
@@ -113,137 +98,84 @@ private async Task ExecuteJobAsync(JobInfo jobInfo)
 
             try
             {
-                try
-                {
-                    if (jobInfo.CancelRequested && !jobCancellationToken.IsCancellationRequested)
-                    {
-                        // For cancelled job, try to execute it for potential cleanup.
-                        jobCancellationToken.Cancel();
-                    }
-
-                    var progress = new Progress<string>((result) =>
-                    {
-                        jobInfo.Result = result;
-                    });
-
-                    var runningJob = Task.Run(() => job.ExecuteAsync(jobInfo, progress, jobCancellationToken.Token));
-                    _activeJobsNeedKeepAlive[jobInfo.Id] = () => KeepAliveSingleJobAsync(jobInfo, jobCancellationToken);
-
-                    jobInfo.Result = await runningJob;
-                }
-                catch (RetriableJobException ex)
-                {
-                    _logger.LogError(ex, "Job {JobId} failed with retriable exception.", jobInfo.Id);
-
-                    // Not complete the job for retriable exception.
-                    return;
-                }
-                catch (JobExecutionException ex)
+                if (jobInfo.CancelRequested)
                 {
-                    _logger.LogError(ex, "Job {JobId} failed.", jobInfo.Id);
-                    jobInfo.Result = JsonConvert.SerializeObject(ex.Error);
-                    jobInfo.Status = JobStatus.Failed;
-
-                    try
-                    {
-                        await _queueClient.CompleteJobAsync(jobInfo, ex.RequestCancellationOnFailure, CancellationToken.None);
-                    }
-                    catch (Exception completeEx)
-                    {
-                        _logger.LogError(completeEx, "Job {JobId} failed to complete.", jobInfo.Id);
-                    }
-
-                    return;
+                    // For cancelled job, try to execute it for potential cleanup.
+                    jobCancellationToken.Cancel();
                 }
-                catch (Exception ex)
-                {
-                    _logger.LogError(ex, "Job {JobId} failed.", jobInfo.Id);
 
-                    object error = new { message = ex.Message };
-                    jobInfo.Result = JsonConvert.SerializeObject(error);
-                    jobInfo.Status = JobStatus.Failed;
-
-                    try
-                    {
-                        await _queueClient.CompleteJobAsync(jobInfo, false, CancellationToken.None);
-                    }
-                    catch (Exception completeEx)
-                    {
-                        _logger.LogError(completeEx, "Job {JobId} failed to complete.", jobInfo.Id);
-                    }
+                var progress = new Progress<string>((result) => { jobInfo.Result = result; });
+
+                var runningJob = useHeavyHeartbeats
+                               ? _queueClient.ExecuteJobWithHeartbeats(
```

Comment:
Naming wise would it be better to change this to ExecuteJobWithHeavyHeatbeatsAsync to distinguish it from the other method name? Otherwise, it is just an overloaded method and not entirely clear on the difference between them.